users/vieting/jobs/nbest.py

```diff
@@ -0,0 +1,117 @@
+            "cpu": 1,
+            "mem": 4,
+        }
+
+    def tasks(self):
+        yield Task("create_files", mini_task=True)
+        yield Task("run", resume="run", rqmt=self.rqmt, args=range(1, self.concurrent + 1))
+        if self.concurrent > 1:
+            yield Task("merge", mini_task=True, tries=2)  # 2 tries to catch fs problems crashing the job
+
+    def create_files(self):
+        self.write_config(self.config, self.post_config, "nbest_from_lattice.config")
+        self.write_run_script(self.exe, "nbest_from_lattice.config")
+
+    def run(self, task_id):
+        log_file = self.out_log_file if self.concurrent <= 1 else self.out_log_file[task_id]
+        self.run_script(task_id, log_file)
+        if self.concurrent <= 1:
+            shutil.move("nbest.1", self.out_nbest_file.get_path())
+
+    def merge(self):
+        with open(self.out_nbest_file.get_path(), "wt") as out:
+            for t in range(1, self.concurrent + 1):
+                assert os.path.getsize("nbest.%d" % t) > 0, "Empty File nbest.%d, maybe restart merge" % t
+                with open("nbest.%d" % t, "rt") as f:
+                    shutil.copyfileobj(f, out)
+
+    @classmethod
+    def create_config(
+        cls,
+        crp,
+        lattice_cache,
+        n,
+        parallelize,
+        extra_config,
+        extra_post_config,
+        **kwargs,
+    ):
+        config, post_config = rasr.build_config_from_mapping(
+            crp,
+            {
+                "corpus": "flf-lattice-tool.corpus",
+                "lexicon": "flf-lattice-tool.lexicon",
+            },
+            parallelize=parallelize,
+        )
+        # segment
+        config.flf_lattice_tool.network.initial_nodes = "segment"
+        config.flf_lattice_tool.network.segment.type = "speech-segment"
+        config.flf_lattice_tool.network.segment.links = "0->archive-reader:1 0->dump-n-best:1"
+
+        # read lattice
+        config.flf_lattice_tool.network.archive_reader.type = "archive-reader"
+        config.flf_lattice_tool.network.archive_reader.links = "reduce-scores"
+        config.flf_lattice_tool.network.archive_reader.format = "flf"
+        config.flf_lattice_tool.network.archive_reader.path = lattice_cache
+        config.flf_lattice_tool.network.archive_reader.flf.append.keys = "confidence"
+        config.flf_lattice_tool.network.archive_reader.flf.append.confidence.scale = 0.0
+
+        # dump n-best
+        config.flf_lattice_tool.network.reduce_scores.type = "reduce"
+        config.flf_lattice_tool.network.reduce_scores.keys = "am lm"
+        config.flf_lattice_tool.network.reduce_scores.links = "n-best"
+
+        config.flf_lattice_tool.network.n_best.type = "n-best"
+        config.flf_lattice_tool.network.n_best.n = n
+        config.flf_lattice_tool.network.n_best.remove_duplicates = True
+        config.flf_lattice_tool.network.n_best.ignore_non_words = True
```

Comment:
One could also make these configurable.

Reply:
are the non-words even dumped by the "dump-n-best" node? otherwise it would just be duplicate word sequences in the output file.

Reply:
From the documentation of `n-best`:

> If the "ignore-non-word" option is activated, then two hypotheses only differing in non-words are considered duplicates.

So I would guess that the dump node includes non-words and that the `ignore-non-words` option only comes into play when `remove-duplicates` is active.

Reply:
So just keep both configurable as it is now?